src/compiler/abepi.rs

```diff
@@ -323,12 +365,55 @@ impl<F: From<u64> + Into<u32> + Clone, V: Clone> CompilationUnit<F, V> {
 
     fn compile_statement_if_then_else(
         &self,
-        _dsym: DebugSymRef,
-        _cond: Expression<F, V>,
-        _when_true: Statement<F, V>,
-        _when_false: Statement<F, V>,
+        dsym: DebugSymRef,
+        cond: Expression<F, V>,
+        when_true: Statement<F, V>,
+        when_false: Statement<F, V>,
     ) -> Vec<CompilationResult<F, V>> {
-        todo!()
+        assert!(cond.is_logic());
+
+        // if A then assert B else assert C
+        // this is equivalent to (if A then assert B) and (if not A then assert C)
+        // this will be equivalent to (not (A and not B)) and (not (not A and not C))
+
+        // Using cond only as OneZero 0F, 1T
+        // For the OneZero result and we can write this as
+        // (1 - (cond * (1-when_true.one_zero))) * (1 - ((1-cond) * (1-when_false.one_zero)))
+        // For the AntiBooly result we can do
+        // cond * when_true.anti_booly * (1-cond) * when_false.anti_booly
```

Comment:
@rutefig the explanation of the anti-booly has a * where in the code there is a +.

Also, if you are reusing the other code, you don't need to convert to formula the reuse. The only thing you are doing outside the reuse, is the AND of `(if A then assert B) and (if not A then assert C)`. If you are implementing it like this, you only need to explain how you are doing the AND.